actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -139,6 +141,31 @@ def negotiate_mime(order)
         order.include?(Mime::ALL) ? format : nil
       end
 
+      class VariantInquirer # :nodoc:
+        delegate :each, :empty?, to: :@variants
+
+        def initialize(variants = [])
+          @variants = variants
+        end
+
+        def any?(*candidates)
+          (@variants & candidates).any?
+        end
+
+        def to_ary
+          @variants
+        end
+
+        private
+          def method_missing(name, *args)
+            if name[-1] == '?'
+              any? name[0..-2].to_sym
```

Comment:
How does this work? `name[0..-2]` will return an array, and arrays don't respond to `to_sym`. Did you mean to `.map(&:to_sym)`?